packages/angular/projects/ui-angular/src/lib/primitives-poc/icon/icon.component/icon.component.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Input,
+  ViewEncapsulation,
+} from '@angular/core';
+import { AmplifyBasePrimitiveComponent } from '../../base-primitive/components/base-primitive.component';
+
+const $COMPONENT_SELECTOR: string = 'amplify-icon';
+@Component({
+  selector: $COMPONENT_SELECTOR,
+  templateUrl: 'icon.component.html',
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AmplifyIconComponent extends AmplifyBasePrimitiveComponent {
+  /** Input Prop for style changes  */
+
+  @Input() pathData: string;
+  @Input() viewBoxValue: string = '';
+
+  public _viewBox: string = '';
+  @Input()
+  public set viewBox(viewBox: string) {
+    let viewBoxObj = JSON.parse(viewBox);
+    this._viewBox = ' 0 0 ' + viewBoxObj.width + ' ' + viewBoxObj.height;
```

Comment:
Instead of JSON parsing, do you think we can accept an object directly?

Reply:
https://github.com/aws-amplify/amplify-ui/pull/2436/commits/c48df471ca3f5621ef92d7cd7b49b508629a3018